src/utils/hooks/useGetFullApplicationStructure.tsx

```diff
@@ -134,13 +150,21 @@ const useGetFullApplicationStructure = (structure: FullStructure, firstRunValida
       evaluationParameters,
       true
     )
-    const results = await Promise.all([isEditable, isRequired, isVisible])
+    const isValid =
+      shouldProcessValidation || firstRunProcessValidation
+        ? evaluateExpressionWithFallBack(element.validationExpression, evaluationParameters, false)
+        : new Promise(() => responseObject[element.code]?.isValid)
```

Comment:
should this resolve ? i.e. `new Promise(resolve => resolve(responseObject[element.code]?.isValid))`

Reply:
I was trying to make it so that it's the type/state as what `isValid` is in the first case -- i.e. unresolved Promise. All promises for these four variables are resolved in the subsequent `const results = await Promise.all([isEditable, isRequired, isVisible, isValid])` line. Basically, this just setting the value of `isValid` to the same as it previously was, but wrapped in a Promise so it's consistent with the value if its getting evaluated. Does that make sense?

Reply:
As our chat

Reply:
Cool, I updated to :
`async () => responseObject[element.code]?.isValid` as per suggestion. That's essentially what I was meaning in the first place. Cheers.